modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/type/AutoDownloadLimitSetting.kt

```diff
@@ -50,10 +50,12 @@ enum class AutoDownloadLimitSetting(
         fun fromPreferenceString(stringValue: String): AutoDownloadLimitSetting {
             return try {
                 val intValue = stringValue.toInt()
-                entries.firstOrNull { it.preferenceInt == intValue } ?: TWO_LATEST_EPISODE
+                entries.firstOrNull { it.id == intValue } ?: TWO_LATEST_EPISODE
```

Comment:
I think it would be better to change this method to return `AutoDownloadLimitSetting?` instead. The current default value, `TWO_LATEST_EPISODES`, mixes different use cases and could lead to potential errors when we need different behaviors.

I suggest removing the default value and determining the appropriate setting at the call sites instead.